compat/mingw.c

```diff
@@ -383,6 +383,21 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+static int do_unlink(const char *pathname, const wchar_t *wpathname,
+		     int (*unlinker)(const wchar_t *))
+{
+	char temppath[MAX_PATH];
+	wchar_t wtemppath[MAX_LONG_PATH];
+	strncpy(temppath, pathname, sizeof(temppath) - 8);
+	strcat(temppath, ".XXXXXX");
```

Comment:
In case we leave temporary files behind (e.g. Git dies between the renaming and unlinking), it should add a string which allows us to identify the origin of those files later. Maybe ".unlink_XXXXXX"?

Reply:
Done